### PR TITLE
Fix NPE thrown from semantic-api for `iterator().next()`

### LINKED ISSUE #29017: NPE thrown from semantic-api for iterator().next()
**Description:**
When trying to create variable for the below snippet it fails with a NPE from semantic-api.
```
public function hello() {
    int[] x = [1, 2, 3];
    x.iterator().next();
}
```

Error thrown;
```
java.lang.NullPointerException
	at io.ballerina.compiler.api.impl.symbols.BallerinaSymbol.<init>(BallerinaSymbol.java:59)
	at io.ballerina.compiler.api.impl.symbols.BallerinaRecordFieldSymbol.<init>(BallerinaRecordFieldSymbol.java:57)
	at io.ballerina.compiler.api.impl.symbols.BallerinaRecordTypeSymbol.fieldDescriptors(BallerinaRecordTypeSymbol.java:62)
	at io.ballerina.compiler.api.impl.symbols.BallerinaRecordTypeSymbol.signature(BallerinaRecordTypeSymbol.java:106)
	at io.ballerina.compiler.api.impl.symbols.BallerinaUnionTypeSymbol.getSignatureForUnion(BallerinaUnionTypeSymbol.java:128)
	at io.ballerina.compiler.api.impl.symbols.BallerinaUnionTypeSymbol.signature(BallerinaUnionTypeSymbol.java:96)
	at org.ballerinalang.langserver.common.utils.FunctionGenerator.generateTypeDefinition(FunctionGenerator.java:67)
	at org.ballerinalang.langserver.codeaction.CodeActionUtil.getPossibleTypes(CodeActionUtil.java:306)
	at....
```
**Steps to reproduce:**

**Affected Versions:**
2.0.0-SLP3-SNAPSHOT

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #29017

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by pubudu91** [comment]
Will it be possible to add a test for this as well?

**Comment by sanjana** [comment]
> Will it be possible to add a test for this as well?

@pubudu91 yes, sure. Added a `Create Variable Code Action` test case as well.

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java [hunk 1]
```diff
@@ -90,6 +94,41 @@ public void setup() {
...
+    @Test(dataProvider = "RecordFieldImports")
+    public void testRecordFieldSymbol(int line, int col, String fieldName, TypeDescKind typeKind, boolean isOptional,
+                                      boolean hasDefaultValue, String signature) {
+        Project project = BCompileUtil.loadProject("test-src/symbol_lookup_with_record_field_imports_test.bal");
+        Package currentPackage = project.currentPackage();
+        ModuleId defaultModuleId = currentPackage.getDefaultModule().moduleId();
+        Document srcFile = getDocumentForSingleSource(project);
+
+        PackageCompilation packageCompilation = currentPackage.getCompilation();
+        SemanticModel model = packageCompilation.getSemanticModel(defaultModuleId);
```
Comment (pubudu91) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java:107]: There are a couple of util methods to simplify this. See the following for example: https://github.com/ballerina-platform/ballerina-lang/blob/f1515f27806ad0d3b239f852ad4994ec25c434ed/tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/ClassSymbolTest.java#L59

Comment (sanjana) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/SymbolBIRTest.java]: Simplified.

**Review by pubudu91 — APPROVED** [review]

**Pull request merged**